inference-engine/src/cldnn_engine/cldnn_graph.cpp

```diff
@@ -252,34 +316,193 @@ InferenceEngine::ICNNNetwork::Ptr CLDNNGraph::GetExecGraphInfoByPrimitivesInfo(s
+            } else {
+                size_t port = 0;
+                for (auto& usr_id : user_ids) {
+                    auto usr_it = std::find_if(primitives_info.begin(),
+                                           primitives_info.end(),
+                                           [&](cldnn::primitive_info& entry) {
+                                                return entry.original_id == usr_id;
+                                            });
+                    if (usr_it == primitives_info.end())
+                        continue;
+
+                    return_node->set_output_type(port,
+                                                details::convertPrecision(desc.getPrecision()),
+                                                ngraph::PartialShape(desc.getDims()));
+                    port++;
+                }
             }
         }
 
-        layer->params[ExecGraphInfoSerialization::ORIGINAL_NAMES] = concat_strings(originalNames, ',');
-        layer->params[ExecGraphInfoSerialization::IMPL_TYPE] = prim_info.kernel_id;
-        layer->params[ExecGraphInfoSerialization::OUTPUT_PRECISIONS] = layer->precision.name();
-        std::string exec_time = "not_executed";
-        if (perfMap.find(prim_info.original_id) != perfMap.end()) {
-            auto perfCounter = perfMap.at(prim_info.original_id).second;
-            if (perfCounter.num > 0) {
-                exec_time = std::to_string(perfCounter.realTime_avg());
+        for (auto&& kvp : extract_primitive_metadata(prim_info)) {
+            if (kvp.first == "layerName") {
+                return_node->set_friendly_name(kvp.second);
+                if (is_output)
+                    results.back()->set_friendly_name(kvp.second + "_result");
+            } else {
+                auto variant = std::make_shared<::ngraph::VariantWrapper<std::string>>(kvp.second);
+                return_node->get_rt_info()[kvp.first] = variant;
+                if (is_output)
+                    results.back()->get_rt_info()[kvp.first] = variant;
             }
         }
+        
+        nodes.push_back(return_node);
+        node2layer[prim_info.original_id] = return_node;
+    };
+    
+    removeMutableDataInfo(primitives_info);
+    for (auto& pi : primitives_info) {
+        // Skip const inputs
+        if (pi.type_id == "data") {
+            continue;
+        }
+
+        // Skip mutable_data
+        if (pi.type_id == "mutable_data" &&
+            pi.c_dependencies.size() == 1 &&
+            !pi.c_users.empty()) {
+            continue;
+        }
+
+        create_ngraph_node(pi);        
+    }
+
+    ngraph::op::GenericIE::DisableReshape reshape(nodes);
+    auto function = std::make_shared<ngraph::Function>(results, params, "runtime_gpu_graph");
+    InferenceEngine::CNNNetwork net(function);
+    return net;
+}
+
+InferenceEngine::ICNNNetwork::Ptr CLDNNGraph::GetExecGraphInfoByPrimitivesInfo(std::vector<cldnn::primitive_info>& primitives_info,
```

Comment:
Why ngraph version can't be used for intermediate stages?

Reply:
Fixed - now only the nGraph implementation exists.